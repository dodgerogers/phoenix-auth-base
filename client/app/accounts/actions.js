import { actionTypes } from './constants';

export const getCurrentUserRequest = () => ({
  type: actionTypes.GET_CURRENT_USER_REQUEST,
  accessToken,
});

export const getCurrentUserSuccess = user => ({
  type: actionTypes.GET_CURRENT_USER_SUCCESS,
  user,
});

export const getCurrentUserFailure = () => ({
  type: actionTypes.GET_CURRENT_USER_FAILURE,
});

export const getCurrentUserProfilesRequest = () => ({
  type: actionTypes.GET_CURRENT_USER_PROFILES_REQUEST,
  accessToken,
});

export const getCurrentUserProfilesSuccess = profiles => ({
  type: actionTypes.GET_CURRENT_USER_PROFILES_SUCCESS,
  profiles,
});

export const getCurrentUserProfilesFailure = () => ({
  type: actionTypes.GET_CURRENT_USER_PROFILES_FAILURE,
});

export const setCurrentUserProfile = (profileId) => ({
  type: actionTypes.SET_CURRENT_USER_PROFILE,
  profileId,
})
