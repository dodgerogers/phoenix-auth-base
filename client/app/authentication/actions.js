import { actionTypes } from './constants';

export const verifyTokenRequest = accessToken => ({
  type: actionTypes.VERIFY_TOKEN_REQUEST,
  accessToken,
});

export const verifyTokenSuccess = () => ({
  type: actionTypes.VERIFY_TOKEN_SUCCESS,
});

export const verifyTokenFailure = () => ({
  type: actionTypes.VERIFY_TOKEN_FAILURE,
});

export const refreshTokenRequest = () => ({
  type: actionTypes.REFRESH_TOKEN_REQUEST,
});

export const refreshTokenSuccess = () => ({
  type: actionTypes.REFRESH_TOKEN_SUCCESS,
});

export const refreshTokenFailure = () => ({
  type: actionTypes.REFRESH_TOKEN_FAILURE,
});

export const refreshTokenRequestCancelled = () => ({
  type: actionTypes.REFRESH_TOKEN_REQUEST_CANCELLED,
});

export const authenticateWithStoredToken = () => ({
  type: actionTypes.AUTHENTICATE_WITH_STORED_TOKEN_REQUEST,
});

export const authenticateFailure = error => ({
  type: actionTypes.AUTHENTICATE_FAILURE,
  error,
});

export const signOutRequest = () => ({
  type: actionTypes.SIGN_OUT_REQUEST,
});

export const signOutSuccess = () => ({
  type: actionTypes.SIGN_OUT_SUCCESS,
});

export const signOutFailure = () => ({
  type: actionTypes.SIGN_OUT_FAILURE,
});

export const signInSuccess = response => ({
  type: actionTypes.SIGN_IN_SUCCESS,
  accessToken: response.data.accessToken,
});

export const registerSuccess = (_result, _dispatch, { values }) => ({
  type: actionTypes.REGISTER_SUCCESS,
  formValues: values,
});

export const registerFailure = error => ({
  type: actionTypes.REGISTER_FAILURE,
  error,
});

export const confirmationSuccess = response => ({
  type: actionTypes.CONFIRMATION_SUCCESS,
  accessToken: response.data.accessToken,
});

export const confirmationFailure = (error) => ({
  type: actionTypes.CONFIRMATION_FAILURE,
  error,
});

export const resendConfirmationSuccess = (_result, _dispatch, { values }) => ({
  type: actionTypes.RESEND_CONFIRMATION_SUCCESS,
  formValues: values,
});

export const resendConfirmationFailure = (error) => ({
  type: actionTypes.RESEND_CONFIRMATION_FAILURE,
  error,
});

export const passwordResetRequestSuccess = (_result, _dispatch, { values }) => ({
  type: actionTypes.PASSWORD_RESET_REQUEST_SUCCESS,
  formValues: values,
});

export const resetPasswordSuccess = (_result, _dispatch, { values }) => ({
  type: actionTypes.RESET_PASSWORD_SUCCESS,
  formValues: values,
});
