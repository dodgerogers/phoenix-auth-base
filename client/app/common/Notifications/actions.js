import { actionTypes } from './constants';


export const notify = (message, areaID) => ({
  type: actionTypes.NOTIFY,
  id: areaID,
  notification: {
    level: 'success',
    message,
  },
});

export const destroy = (notification, areaID) => ({
  type: actionTypes.DESTROY,
  id: areaID,
  notification,
});
