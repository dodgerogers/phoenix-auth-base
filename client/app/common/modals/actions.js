import { actionTypes } from './constants';

export function showModal(id) {
  return {
    type: actionTypes.SHOW_MODAL,
    data: { id },
  };
}

export function hideModal(id) {
  return {
    type: actionTypes.HIDE_MODAL,
    data: { id },
  };
}
