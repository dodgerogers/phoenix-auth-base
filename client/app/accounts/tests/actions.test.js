import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as AccountActions from '../actions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('AccountActions', () => {
  describe('getCurrentUser', () => {
    it('getCurrentUserRequest matches snapshot', () => {
      expect(AccountActions.getCurrentUserRequest()).toMatchSnapshot();
    });

    it('getCurrentUserSuccess matches snapshot', () => {
      expect(AccountActions.getCurrentUserSuccess()).toMatchSnapshot();
    });

    it('getCurrentUserFailure matches snapshot', () => {
      expect(AccountActions.getCurrentUserFailure()).toMatchSnapshot();
    });
  });

  describe('getCurrentUserProfiles', () => {
    it('getCurrentUserProfilesRequest matches snapshot', () => {
      expect(AccountActions.getCurrentUserProfilesRequest()).toMatchSnapshot();
    });

    it('getCurrentUserProfilesSuccess matches snapshot', () => {
      expect(AccountActions.getCurrentUserProfilesSuccess([{id: 1}])).toMatchSnapshot();
    });

    it('getCurrentUserProfilesFailure matches snapshot', () => {
      expect(AccountActions.getCurrentUserProfilesFailure()).toMatchSnapshot();
    });
  });
});
