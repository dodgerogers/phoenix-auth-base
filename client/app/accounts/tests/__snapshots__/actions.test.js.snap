// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountActions getCurrentUser getCurrentUserFailure matches snapshot 1`] = `
Object {
  "type": "GET_CURRENT_USER_FAILURE",
}
`;

exports[`AccountActions getCurrentUser getCurrentUserRequest matches snapshot 1`] = `
Object {
  "type": "GET_CURRENT_USER_REQUEST",
}
`;

exports[`AccountActions getCurrentUser getCurrentUserSuccess matches snapshot 1`] = `
Object {
  "type": "GET_CURRENT_USER_SUCCESS",
  "user": undefined,
}
`;

exports[`AccountActions getCurrentUserProfiles getCurrentUserProfilesFailure matches snapshot 1`] = `
Object {
  "type": "GET_CURRENT_USER_PROFILES_FAILURE",
}
`;

exports[`AccountActions getCurrentUserProfiles getCurrentUserProfilesRequest matches snapshot 1`] = `
Object {
  "type": "GET_CURRENT_USER_PROFILES_REQUEST",
}
`;

exports[`AccountActions getCurrentUserProfiles getCurrentUserProfilesSuccess matches snapshot 1`] = `
Object {
  "profiles": Array [
    Object {
      "id": 1,
    },
  ],
  "type": "GET_CURRENT_USER_PROFILES_SUCCESS",
}
`;
